bootspec/src/generation.rs

```diff
@@ -56,11 +66,12 @@ mod tests {
         ],
         "label": "NixOS 21.11.20210810.dirty (Linux 5.15.30)",
         "toplevel": "/nix/store/xxx-nixos-system-xxx",
-        "specialisation": {}
+        "specialisation": {},
+        "extensions": { "org.test": "hello" }
     }
 }"#;
 
-        let from_json: Generation = serde_json::from_str(&json).unwrap();
+        let from_json: Generation<TestExtension> = serde_json::from_str(&json).unwrap();
```

Comment:
If we do keep the generic, I'd rather see this become its own test (a `valid_v1_json` test, and a `valid_v1_json_with_typed_extension` test).

Reply:
Sure thing :)

Reply:
I would also add some tests regarding "missing" fields

Reply:
More tests are always welcome!